src/diffusers/pipelines/stable_audio/pipeline_stable_audio.py

```diff
@@ -0,0 +1,776 @@
+
+        Args:
+            prompt (`str` or `List[str]`, *optional*):
+                prompt to be encoded
+            audio_start_in_s (`float` or `List[float]`, *optional*):
+                Seconds indicating the start of the audios, to be encoded.
+            audio_end_in_s (`float` or `List[float]`, *optional*)
+                Seconds indicating the end of the audios, to be encoded.
+            device (`torch.device`):
+                torch device
+            num_waveforms_per_prompt (`int`):
+                number of waveforms that should be generated per prompt
+            do_classifier_free_guidance (`bool`):
+                whether to use classifier free guidance or not
+            negative_prompt (`str` or `List[str]`, *optional*):
+                The prompt or prompts not to guide the audio generation. If not defined, one has to pass
+                `negative_cross_attention_hidden_states` instead. Ignored when not using guidance (i.e., ignored if `guidance_scale` is
+                less than `1`).
+            cross_attention_hidden_states (`torch.Tensor`, *optional*):
+                Pre-computed text embeddings from the T5 model. Can be used to easily tweak text inputs, *e.g.*
+                prompt weighting. If not provided, text embeddings will be computed from `prompt` input argument.
+            negative_cross_attention_hidden_states (`torch.Tensor`, *optional*):
+                Pre-computed negative text embeddings from the T5 model. Can be used to easily tweak text inputs,
+                *e.g.* prompt weighting. If not provided, negative_cross_attention_hidden_states will be computed from
+                `negative_prompt` input argument.
+            attention_mask (`torch.LongTensor`, *optional*):
+                Pre-computed attention mask to be applied to the `cross_attention_hidden_states`. If not provided, attention mask will
+                be computed from `prompt` input argument.
+            negative_attention_mask (`torch.LongTensor`, *optional*):
+                Pre-computed attention mask to be applied to the `negative_cross_attention_hidden_states`. If not provided, attention
+                mask will be computed from `negative_prompt` input argument.
+        Returns:
+            cross_attention_hidden_states (`torch.Tensor`):
+                Text embeddings from the T5 model.
+            attention_mask (`torch.LongTensor`):
+                Attention mask to be applied to the `cross_attention_hidden_states`.
+
+        Example:
+
+        ```python
+        >>> import scipy
+        >>> import torch
+        >>> from diffusers import StableAudioPipeline
+
+        >>> repo_id = "cvssp/audioldm2"
+        >>> pipe = StableAudioPipeline.from_pretrained(repo_id, torch_dtype=torch.float16)
+        >>> pipe = pipe.to("cuda")
+
+        >>> # Get text embedding vectors
+        >>> cross_attention_hidden_states, attention_mask = pipe.encode_prompt(
```

Comment:
(nit): This should be replaced with the method that's being implemented here: `encode_prompt_and_seconds()`.